### DISCUSSION TITLE
Is there a way to require all nodes run specific docker engine version?

### DISCUSSION THREAD

**Discussion by YuryHrytsuk** [moby/moby#45885] [General]
This is to avoid bugs in case docker swarm orchestrates nodes that run different docker versions

**Comment by thaJeztah** [comment]
No, there's no built-in constraint on version of the engine that's running, but you could probably handle this through node-labels, and use those for scheduling; https://docs.docker.com/engine/swarm/services/#placement-constraints


In general, it's not recommended to run a mixed-version swarm cluster, and it's best to (after verifying in a test-environment before rolling out to production) update all nodes in the cluster to be on the same version.
